src/main/java/org/fenixedu/academic/ui/spring/controller/StatuteTypeManagement/StatuteTypeController.java

```diff
@@ -0,0 +1,190 @@
+import org.fenixedu.academic.ui.spring.AcademicConfigurationsController;
+import org.fenixedu.bennu.core.domain.exceptions.DomainException;
+import org.fenixedu.bennu.spring.portal.SpringFunctionality;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import pt.ist.fenixframework.Atomic;
+
+@SpringFunctionality(app = AcademicConfigurationsController.class, title = "label.title.StatuteTypeManagement",
+        accessGroup = "#managers")
+@RequestMapping("/academic-configurations/statutetypemanagement/statutetype")
+public class StatuteTypeController extends AcademicBaseController {
+    @RequestMapping
+    public String home(Model model) {
+        //this is the default behaviour, for handling in a Spring Functionality
+        return "redirect:/academic-configurations/statutetypemanagement/statutetype/";
+    }
+
+    private StatuteType getStatuteType(Model m) {
+        return (StatuteType) m.asMap().get("statuteType");
+    }
+
+    private void setStatuteType(StatuteType statuteType, Model m) {
+        m.addAttribute("statuteType", statuteType);
+    }
+
+    @Atomic
+    public void deleteStatuteType(StatuteType statuteType) {
+        statuteType.delete();
+    }
+
+    @RequestMapping(value = "/")
+    public String search(@RequestParam(value = "code", required = false) java.lang.String code, Model model) {
+        List<StatuteType> searchstatutetypeResultsDataSet = filterSearchStatuteType(code);
+
+        //add the results dataSet to the model
+        model.addAttribute("searchstatutetypeResultsDataSet", searchstatutetypeResultsDataSet);
+        return "academic-configurations/statutetypemanagement/statutetype/search";
+    }
+
+    private List<StatuteType> collectSearchStatuteTypeDataSet() {
+        return new ArrayList<StatuteType>(StatuteType.readAll());
+    }
+
+    private List<StatuteType> filterSearchStatuteType(java.lang.String code) {
+
+        return collectSearchStatuteTypeDataSet()
+                .stream()
+                .filter(statuteType -> code == null
+                        || code.length() == 0
+                        || (statuteType.getCode() != null && statuteType.getCode().length() > 0 && statuteType.getCode()
+                                .toLowerCase().contains(code.toLowerCase()))).collect(Collectors.toList());
+    }
+
+    @RequestMapping(value = "/read/{oid}")
+    public String read(@PathVariable("oid") StatuteType statuteType, Model model) {
+        setStatuteType(statuteType, model);
+        return "academic-configurations/statutetypemanagement/statutetype/read";
+    }
+
+    @RequestMapping(value = "/delete/{oid}")
+    public String delete(@PathVariable("oid") StatuteType statuteType, Model model) {
```

Comment:
This seems like too much of a delicate operation to be performed in a GET request.